src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -219,15 +221,35 @@ def get_artifact_bundles_containing_debug_id(
 def get_artifact_bundles_containing_url(
     project: Project, release_name: str, dist_name: str, url: str
 ) -> Set[Tuple[int, datetime]]:
-    return set(
-        ArtifactBundleIndex.objects.filter(
-            organization_id=project.organization.id,
-            release_name=release_name,
-            dist_name=dist_name,
-            url=url,
-            artifact_bundle__projectartifactbundle__project_id=project.id,
-        ).values_list("artifact_bundle_id", "date_added")[:1]
+    filter = Q(
+        organization_id=project.organization.id,
+        release_name=release_name,
+        dist_name=dist_name,
+        url=url,
+        artifact_bundle__projectartifactbundle__project_id=project.id,
     )
+    found_bundles = list(
+        ArtifactBundleIndex.objects.filter(filter).values_list(
+            "artifact_bundle_id", "date_added", "date_last_modified"
+        )
+        # we want to return the latest one, and only care *if* there are any others
+        .order_by("-date_last_modified")[:2]
+    )
+
+    bundles = set()
+    if len(found_bundles) > 0:
+        artifact_bundle_id, date_added, date_last_modified = found_bundles[0]
+        bundles.add((artifact_bundle_id, date_added))
+        if len(found_bundles) > 1:
+            metrics.incr("artifact_bundle_indexing.read_with_conflict")
+            if options.get("sourcemaps.artifact-bundles.enable_index_cleanup") == 1.0:
+                # here, we want to delete all the duplicated keys with an older `date_last_modified`:
+                num_deleted, _per_model = ArtifactBundleIndex.objects.filter(
+                    filter & Q(date_last_modified__lt=date_last_modified)
```

Comment:
Here you would have to use the condition that also covers in case the dates are equal, thus you should compare the artifact bundle id in that case.